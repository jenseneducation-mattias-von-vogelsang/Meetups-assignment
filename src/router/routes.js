//******************************/
//*** ROUTES FOR VUE ROUTER ***/
//****************************/
const routes = [
  {
    path: '/',
    name: 'Home',
    component: () => import('../views/Home/Home.vue'),
  },
  {
    path: '/Account',
    name: 'Account',
    component: () => import('../views/Account/Account.vue'),
    meta: {
      requiresAuth: true,
    },
  },
  {
    path: '/CreateMeetup',
    name: 'CreateMeetup',
    component: () => import('../views/CreateMeetup/CreateMeetup.vue'),
    meta: {
      requiresAuth: true,
    },
  },
  {
    path: '/Login',
    name: 'Login',
    component: () => import('../views/Login/Login.vue'),
    meta: {
      requiresGuest: true,
    },
  },
  {
    path: '/Register',
    name: 'Register',
    component: () => import('../views/Register/Register.vue'),
    meta: {
      requiresGuest: true,
    },
  },
  {
    path: '/MeetupDetails/:id',
    name: 'MeetupDetails',
    component: () => import('../views/MeetupDetails/MeetupDetails.vue'),
  },
];

export default routes;
