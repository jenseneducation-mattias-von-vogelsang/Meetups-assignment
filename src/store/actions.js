//***********************/
//*** GLOBAL ACTIONS ***/
//*********************/
const actions = {
  async loginThisUser(context, value) {
    let user = {
      username: value.username,
      password: value.password,
    };
    await context.dispatch('login', user);
  },
  async registerThisUser(context, value) {
    let user = {
      username: value.username,
      password: value.password,
      confirm_password: value.confirm_password,
      email: value.email,
      name: value.name,
    };
    await context.dispatch('register', user);
  },
  async sendThisReview(context, id, text) {
    await context.dispatch('sendReview', id, text);
  },
};

export default actions;
