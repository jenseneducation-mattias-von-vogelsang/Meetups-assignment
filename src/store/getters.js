import { getField } from 'vuex-map-fields';
//***********************/
//*** GLOBAL GETTERS ***/
//*********************/
const getters = {
  getField,
  isAttending: (state, rootGetters) => {
    let name = rootGetters.user.name;
    let checkAttendance = rootGetters.clickedMeetup.attendees.filter(
      (item) => item === name
    );
    if (checkAttendance.length == 1) {
      return true;
    } else {
      return false;
    }
  },
  alreadyReviewed: (state, rootGetters) => {
    let name = rootGetters.user.username;
    let checkReview = rootGetters.clickedMeetup.reviews.filter(
      (item) => item.username === name
    );
    if (checkReview.length == 1) {
      return true;
    } else {
      return false;
    }
  },
  getLocation: (state) => state.meetupForm.location,
};

export default getters;
