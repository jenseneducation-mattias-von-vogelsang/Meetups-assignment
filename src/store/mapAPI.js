import axios from 'axios';

//**************************************/
//*** MAP & LOCATION API OPERATIONS ***/
//************************************/

//TODO: fix api coordinate operations
const api = axios.create({
  baseURL: 'https://api.mapbox.com/geocoding/v5/mapbox.places/',
});
const token = process.env.VUE_APP_MAPBOX_API_KEY;

const actions = {
  async geocodeLocation({ commit }, address) {
    let res = await api.get(
      `${encodeURIComponent(address)}.json?access_token=${token}`
    );
    if (res.data.features.length > 0) {
      await commit('geocoding_success', res.data.features[0].geometry);
    }
    if (res.data.features.length == 0) {
      await commit('geocoding_fail');
    }
    return res;
  },
};

export default {
  actions,
};
