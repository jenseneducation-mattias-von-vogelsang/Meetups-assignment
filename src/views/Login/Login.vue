<template>
  <div>
    <LoginForm />
    <router-link to="/Register" tag="button"
      ><i>Not registered yet?</i></router-link
    >
  </div>
</template>
<script>
import LoginForm from '@/components/LoginForm/LoginForm.vue';
export default {
  name: 'Login',
  components: {
    LoginForm,
  },
};
</script>
