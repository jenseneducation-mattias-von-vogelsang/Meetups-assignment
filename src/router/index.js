import Vue from 'vue';
import VueRouter from 'vue-router';
import routes from './routes.js';
// import store from '../store/index.js';

Vue.use(VueRouter);

const router = new VueRouter({
  mode: 'history',
  base: process.env.BASE_URL,
  routes,
});

//************************************/
//*** NAVIGATION GUARDS FOR LOGIN ***/
//**********************************/

//Router navigation guards for login & account page

// router.beforeEach((to, from, next) => {
//   if (to.matched.some((record) => record.meta.requiresAuth)) {
//     if (!store.getters.isLoggedIn) {
//       // Redirect to the Login Page
//       next('/Login');
//     } else {
//       next();
//     }
//   } else if (to.matched.some((record) => record.meta.requiresGuest)) {
//     if (store.getters.isLoggedIn) {
//       // Redirect to the Login Page
//       next('/Account');
//     } else {
//       next();
//     }
//   } else {
//     next();
//   }
// });

export default router;
