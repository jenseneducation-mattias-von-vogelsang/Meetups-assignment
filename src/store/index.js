//****************************************/
//*** IMPORT STORE MODULES & VUE/VUEX ***/
//**************************************/
import Vue from 'vue';
import Vuex from 'vuex';
import actions from './actions.js';
import getters from './getters.js';
import mutations from './mutations.js';
import Auth from './auth.js';
import Meetups from './meetups.js';

Vue.use(Vuex);
//*********************/
//*** GLOBAL STATE ***/
//*******************/
export default new Vuex.Store({
  state: {
    loginForm: {
      username: '',
      password: '',
    },
    registerForm: {
      name: '',
      username: '',
      confirm_password: '',
      email: '',
      password: '',
    },
    newReview: [
      {
        username: '',
      },
    ],
  },
  mutations: mutations,
  actions: actions,
  getters: getters,
  modules: {
    Auth,
    Meetups,
  },
});
