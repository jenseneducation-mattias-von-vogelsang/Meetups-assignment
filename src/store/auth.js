import axios from 'axios';
import router from '../router/index.js';

//****************************/
//*** SET BASEURL FOR API ***/
//**************************/
// TODO: change for deployment of backend server to heroku
const api = axios.create({
  baseURL: 'http://localhost:5000/api',
});

//*****************************/
//*** AUTHENTICATION STATE ***/
//***************************/

const state = {
  token: window.localStorage.getItem('token') || '',
  user: {
    createdMeetups: '',
    date: '',
    email: '',
    name: '',
    username: '',
  },
  status: '',
  error: null,
};

//*******************************/
//*** AUTHENTICATION GETTERS ***/
//*****************************/

const getters = {
  isLoggedIn: (state) => !!state.token,
  authState: (state) => state.status,
  user: (state) => state.user,
  error: (state) => state.error,
};

//*******************************/
//*** AUTHENTICATION ACTIONS ***/
//*****************************/

const actions = {
  //Login action
  async login({ commit }, user) {
    commit('auth_request');
    try {
      let res = await api.post('/users/login', user);
      if (res.data.success) {
        const token = res.data.token;
        const user = res.data.user;
        // Store the token in localStorage
        await commit('set_token', token);
        commit('auth_success', token, user);
        router.push('/Account');
      }
      return res;
    } catch (err) {
      commit('auth_error', err);
    }
  },
  // Register new user
  async register({ commit }, userData) {
    try {
      commit('register_request');
      let res = await api.post('/users/register', userData);
      console.log(res);
      if (res.data.success) {
        await commit('register_success');
        router.push('/Login');
      }
      return res;
    } catch (err) {
      commit('register_error', err);
    }
  },
  // Get the user profile
  async getProfile({ commit }) {
    commit('profile_request');
    api.defaults.headers.common['Authorization'] = state.token;
    let res = await api.get('/users/profile');
    commit('user_profile', res.data.user);
    return res;
  },
  // Logout the user
  async logout({ commit }) {
    window.localStorage.removeItem('token');
    commit('logout');
    delete axios.defaults.headers.common['Authorization'];
    router.push('/Login');
    return;
  },
};

//*********************************/
//*** AUTHENTICATION MUTATIONS ***/
//*******************************/

const mutations = {
  auth_request(state) {
    state.error = null;
    state.status = 'Loading..';
  },
  auth_success(state, token, user) {
    state.token = token;
    state.user = user;
    state.status = 'Success';
    state.error = null;
  },
  auth_error(state, err) {
    state.error = err.response.data.msg;
  },
  register_request(state) {
    state.error = null;
    state.status = 'Loading...';
  },
  register_success(state) {
    state.error = null;
    state.status = 'Success';
  },
  register_error(state, err) {
    state.error = err.response.data.msg;
  },
  logout(state) {
    state.error = null;
    state.status = '';
    state.token = '';
    state.user = '';
  },
  profile_request(state) {
    state.status = 'Loading...';
  },
  user_profile(state, user) {
    state.user = user;
  },
  set_token(state, token) {
    window.localStorage.setItem('token', token);
  },
  get_token(state) {
    return state.token;
  },
};

export default {
  state,
  actions,
  mutations,
  getters,
};
