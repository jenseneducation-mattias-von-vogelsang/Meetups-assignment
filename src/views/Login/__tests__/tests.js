import {
  createLocalVue,
  shallowMount,
  mount,
  enableAutoDestroy,
} from '@vue/test-utils';
import Vuex from 'vuex';
import Login from '@/views/Login/Login.vue';
import VueRouter from 'vue-router';
import store from '@/store/index.js';
import router from '@/router/index.js';
import Vue from 'vue';
import Vuetify from 'vuetify';

const localVue = createLocalVue();
localVue.use(VueRouter, Vuex);

//Destroy wrapper after each test
enableAutoDestroy(afterEach);

describe('User navigates to home page', () => {
  let vuetify;

  beforeEach(() => {
    vuetify = new Vuetify();
    Vue.use(Vuetify);
  });
  test('It renders correctly and matches snapshot', async () => {
    const wrapper = mount(Login, {
      store,
      localVue,
      router,
      vuetify,
    });
    expect(wrapper.html()).toMatchSnapshot();
  });
});
