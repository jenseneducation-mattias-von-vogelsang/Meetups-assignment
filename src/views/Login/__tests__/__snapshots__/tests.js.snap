// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`User navigates to home page It renders correctly and matches snapshot 1`] = `
<div class="container d-flex flex-column align-center">
  <div class="spacer"></div>
  <div class="container fill-height container--fluid">
    <div class="row align-center justify-center">
      <div class="col-sm-8 col-md-6 col-12">
        <div class="elevation-12 v-card v-sheet theme--light">
          <header class="gradient v-sheet theme--dark v-toolbar v-toolbar--flat" style="height: 64px;">
            <div class="v-toolbar__content" style="height: 64px;">
              <div class="v-toolbar__title font-weight-bold">Login</div>
              <div class="spacer"></div>
            </div>
          </header>
          <div class="v-card__text">
            <form novalidate="novalidate" class="v-form">
              <div class="v-input theme--light v-text-field v-text-field--placeholder">
                <div class="v-input__prepend-outer">
                  <div class="v-input__icon v-input__icon--prepend"><i aria-hidden="true" class="v-icon notranslate mdi mdi-account theme--light"></i></div>
                </div>
                <div class="v-input__control">
                  <div class="v-input__slot">
                    <div class="v-text-field__slot"><label for="username" class="v-label v-label--active theme--light" style="left: 0px; position: absolute;">Username</label><input autocomplete="username" name="username" required="required" id="username" placeholder="Username" type="text"></div>
                  </div>
                  <div class="v-text-field__details">
                    <div class="v-messages theme--light">
                      <transition-group-stub tag="div" name="message-transition" class="v-messages__wrapper"></transition-group-stub>
                    </div>
                  </div>
                </div>
              </div>
              <div class="v-input theme--light v-text-field v-text-field--placeholder">
                <div class="v-input__prepend-outer">
                  <div class="v-input__icon v-input__icon--prepend"><i aria-hidden="true" class="v-icon notranslate mdi mdi-lock theme--light"></i></div>
                </div>
                <div class="v-input__control">
                  <div class="v-input__slot">
                    <div class="v-text-field__slot"><label for="password" class="v-label v-label--active theme--light" style="left: 0px; position: absolute;">Password</label><input autocomplete="current-password" name="password" required="required" id="password" placeholder="Password..." type="password"></div>
                  </div>
                  <div class="v-text-field__details">
                    <div class="v-messages theme--light">
                      <transition-group-stub tag="div" name="message-transition" class="v-messages__wrapper"></transition-group-stub>
                    </div>
                  </div>
                </div>
              </div>
            </form>
          </div>
          <div class="v-card__actions">
            <div class="spacer"></div> <button type="button" class="font-weight-bold v-btn v-btn--contained theme--dark v-size--default orange" id="loginBtn"><span class="v-btn__content">Login</span></button>
          </div>
        </div>
      </div>
    </div>
  </div> <button class=""><i>Not registered yet?</i></button>
</div>
`;
